worker/codalabworker/fsm.py

```diff
@@ -0,0 +1,64 @@
+import threading
+import time
+
+
+class State(object):
+    """
+    The base class for a state in a finite state machine (FSM).
+    """
+    def update(self):
+        """
+        The update method which is called as part of the primary run loop of a FSM.
+        :return: a tuple of a new state for the FSM and a list of output messages
```

Comment:
Comment is out of date; you just return the new state. 

I'm not sure about transition_noop and transition_done; it seems like given that you're just returning a state from update, `return self` is reasonably clear. I agree that `return None` isn't great, but maybe `return State.DONE` (which is None) is better? I'm just suggesting this because on first read I didn't understand whether transition_noop and transition_done were intended to be overridden by subclasses or called to cause a transition or what. (Their names made it sound to me like they could be callbacks to the state subclass for when certain transitions happened.) 

Generally might want to just add some high-level context about how to use this to the class docstring.